### ISSUE TITLE
Linux GTK: Horizontal scroll bar problem

### ISSUE THREAD

**Issue by funkymaster** [doublecmd/doublecmd#567]
When I use the settings below, the horizontal scroll bar doesn't work if the window size is smaller than the panel would need for all columns.

```
AutoFillColumns>True
AutoSizeColumn>0
```

The left side of the panel is then cut off and the columns cannot be moved.

Please fix.
Many Thanks.

> Double Commander
> Version: 1.1.0 alpha
> Revision: 10333
> Commit: e2aecba
> Build date: 2022/08/28
> Lazarus: 2.2.2.0
> FPC: 3.2.2
> Platform: x86_64-Linux-gtk2
> OS version: Debian bookworm/sid
> 
> Widgetset library: GTK 2.24.33

**Comment by funkymaster** [comment]
It's not just Linux that's affected.
The problem also occurs on windows

> Double Commander
> Version: 1.1.0 alpha
> Revision: 10331
> Commit: f6e99c46c
> Build date: 2022/08/28
> Lazarus: 2.2.3.0
> FPC: 3.2.2
> Platform: x86_64-Win64-win32/win64
> OS version: Windows 7 SP1 x86_64

**Comment by alexx2000** [comment]
No bug here. `AutoFillColumns` makes columns total width equal to panel width. It is a feature of this mode.

**Comment by Skif-off** [comment]
>  It is a feature of this mode. I cannot change this behavior.

If this will done, then "Auto fill columns" will have no sense.

**Comment by funkymaster** [comment]
Yes, I know this is a feature and I really appreciate it.
The behaviour that is undesirable for me only occurs as soon as the panel becomes smaller than the space required for the columns. E.g. if the app is tiled to one side.
By the way, my first post wasn't entirely correct. The columns of the panel are not cut off on the left, but the 2nd column is aligned on the left and the columns that can no longer be displayed are cut off on the right side.

Please do a short test.
Minimise the window width step by step.
Then it can be seen that the files and folder names of the first column (AutoFillColumn) move into the second column until the text of the second column would be overwritten, then the text of the first column is hidden. Meanwhile, the columns on the right side are pushed further and further out of view and cut off.

In my view, the desired behaviour would be the following:
- The text of the first column (AutoFillColumn) is not moved into and does not overlap the second column.
- As soon as the panel size is no longer sufficient for all columns, the `AutoFillColumns` feature is automatically and temporarily deactivated.
- As soon as the panel size is no longer sufficient for all columns, and `AutoFillColumns` is deactivated, a horizontal scroll bar is displayed.

**Comment by Skif-off** [comment]
@funkymaster , maybe somthing like "Minimal size" for "Auto size column"? And if the size ot the first (or last) colum < "Minimal size" then DC will switch "Auto fill columns" to `false`.
But we will have a problem: "Auto fill columns" is a global parameter, for all tabs. Should DC store the state for each tab separately?


@WhiteChairFromIkea
"Auto fill columns" works as, for example, "Fit Image to window" in graphic viewers: in this case, there simply cannot be a scrollbars :) Maybe just move the file panels separator? `60% width / 40% width` instead of 50/50?

**Comment by WhiteChairFromIkea** [comment]
I have just readjusted some settings, as per updated comment below. I will try out new behavior from now on, I see no problems now, when H scrollbar becomes visible :)

**Comment by funkymaster** [comment]
> maybe somthing like "Minimal size" for "Auto size column"? And if the size ot the first (or last) colum < "Minimal size" then DC will switch "Auto fill columns" to `false`.

Yes, that would be an idea. But do we really need a user configuration here or can DC itself determine dynamically if the size is no longer sufficient? Because the required "Minimal size" will most likely be different for each folder displayed. So I think that a hard user configuration will very often not fit.

> But we will have a problem: "Auto fill columns" is a global parameter, for all tabs. Should DC store the state for each tab separately?

Yes, this thought has crossed my mind as well. That seems to be necessary then.
